tools/boilerplate.py

```diff
@@ -169,6 +169,12 @@ def generate_function(name, called_fullname, template, **kwargs):
 
     # Replace self argument.
     params = list(signature.parameters.values())[1:]
+    has_return_value = (
+        signature.return_annotation is not signature.empty and
```

Comment:
I guess I would expect empty return value to be equivalent to `Any`, and therefore to add the return in anyway.

(which might remove the list of exceptions below)

Reply:
Ah, so something like:
```suggestion
    has_return_value = str(signature.return_annotation) != 'None'
```
appears to work.